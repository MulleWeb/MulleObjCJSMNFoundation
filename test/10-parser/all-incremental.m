//
//  main.m
//  archiver-test
//
//  Created by Nat! on 19.04.16.
//  Copyright Â© 2016 Mulle kybernetiK. All rights reserved.
//
#import <MulleObjCJSMNFoundation/MulleObjCJSMNFoundation.h>
//#import "MulleStandaloneObjCFoundation.h"


//
// Dates can not be tested, because we need the POSIX Foundation or the
// equivalent, which provides the NSDateFormatter functionality
//
//
// stolen randomly from https://github.com/jagill/JAGPropertyConverter
//
static char   test_json[] = "[\n"
"   -1849,\n"
"   18.48e+10,\n"
"   true,\n"
"   false,\n"
"   null,\n"
"   { \"key\": \"value\" }\n"
"]";



int   main( int argc, const char * argv[])
{
   MulleJSMNParser   *parser;
   id                    plist;
   NSUInteger            i;
   NSUInteger            length;

   parser = [[MulleJSMNParser new] autorelease];
   length = strlen( test_json);
   for( i = 1; i < length; i++)
   {
      plist = [parser parseBytes:test_json
                          length:strlen( test_json)];
      if( plist)
         break;
   }

   if( ! plist)
   {
      fprintf( stderr, "failed\n");
      return( 1);
   }

   // not so pretty yet
   printf( "%s\n", [[plist description] UTF8String]);
   return( 0);
}
