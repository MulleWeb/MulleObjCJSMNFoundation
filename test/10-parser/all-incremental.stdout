(
    "-1849",
    "1.848e+11",
    1,
    0,
    NSNull,
    {
        key = value;
    }
)
